packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -105,69 +90,100 @@ export class Stage extends cdk.Construct implements cpapi.IStage, cpapi.IInterna
     return this._actions.slice();
   }
 
-  public render(): CfnPipeline.StageDeclarationProperty {
+  public render(): any {
     return {
-      name: this.node.id,
+      name: this.stageName,
       actions: this._actions.map(action => this.renderAction(action)),
     };
   }
 
+  public addAction(action: cpapi.Action): cpapi.IStage {
+    // check for duplicate Actions and names
+    if (this._actions.find(a => a.actionName === action.actionName)) {
+      throw new Error(`Stage ${this.stageName} already contains an Action with name '${action.actionName}'`);
+    }
+
+    this._actions.push(action);
+    if (this.stageIsAttachedToPipeline()) {
+      this.attachActionToPipeline(action);
+    }
+
+    return this;
+  }
+
   public onStateChange(name: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps) {
```

Comment:
This should probably be exposed in `IStage` as well

Reply:
Yep, great catch. Added to `IStage`.